Eastern-Research-Group/mywaterway
# Feature/hmw 13 show restoration protection plans on map

## Related Issues:
* https://jira.epa.gov/browse/HMW-13
* https://jira.epa.gov/browse/HMW-59
* https://jira.epa.gov/browse/HMW-5 - addresses comment about hiding surrounding waterbodies layer when WSIO layer is on

## Main Changes:
* Added GIS data to the map for the `Restoration Plans` tab of the `Restore` panel.
* Added GIS data to the map for the `Protection Projects` accordion item of the `Protect` tab.
* Added attains protection projects data to the right hand list for the `Protection Projects` accordion item of the `Protect` tab.
* Fixed an issue where users could highlight waterbodies on the surrounding waterbodies layer by hovering over them. This caused weird issues where waterbodies would be highlighted even though the waterbodies layer was turned off.
* Added code to hide the surrounding waterbodies layer when the WSIO layer is turned on.

## Steps To Test:
1. Navigate to http://localhost:3000/community/040302020902/restore
2. Click on the `Restoration Plans` tab
3. Verify there are no items in the list with a `Plan Type` of `Protection Approach`.
4. Verify that waterbodies are shown on the map in blue
5. Click on a waterbody and verify a popup loads. The popup will be a list of projects/plans associated with that waterbody
6. Click on the `Protect` panel
7. Expand the `Protection Projects` accordion
8. Verify that all items in the list have a `Plan Type` of `Protection Approach`.
9. Verify that waterbodies are shown on the map in blue
10. Click on a waterbody and verify a popup loads. The popup will be a list of projects/plans associated with that waterbody

## TODO:
- [ ] Need to get EPA input on what to name the toggles (Attains Plans and GRTS Projects) in the `Protection Projects` accordion item of the `Protect` tab, or whether or not these toggles are even needed.
- [ ] Get EPA guidance on what they would like to see in the popups. The `Plan Type` item isn't really necessary on the `Protect` panel.
